api/src/main/java/org/pmiops/workbench/api/UserMetricsController.java

```diff
@@ -13,124 +13,63 @@
 import java.util.stream.Collectors;
 import javax.inject.Provider;
 import org.pmiops.workbench.db.dao.UserRecentResourceService;
 import org.pmiops.workbench.db.dao.WorkspaceDao;
-import org.pmiops.workbench.db.model.DbCohort;
-import org.pmiops.workbench.db.model.DbConceptSet;
 import org.pmiops.workbench.db.model.DbUser;
 import org.pmiops.workbench.db.model.DbUserRecentResource;
 import org.pmiops.workbench.db.model.DbWorkspace;
 import org.pmiops.workbench.firecloud.FireCloudService;
 import org.pmiops.workbench.firecloud.model.FirecloudWorkspaceResponse;
 import org.pmiops.workbench.google.CloudStorageClient;
-import org.pmiops.workbench.model.Cohort;
-import org.pmiops.workbench.model.ConceptSet;
 import org.pmiops.workbench.model.EmptyResponse;
 import org.pmiops.workbench.model.FileDetail;
 import org.pmiops.workbench.model.RecentResourceRequest;
 import org.pmiops.workbench.model.WorkspaceAccessLevel;
 import org.pmiops.workbench.model.WorkspaceResource;
 import org.pmiops.workbench.model.WorkspaceResourceResponse;
-import org.pmiops.workbench.utils.mappers.CommonMappers;
-import org.pmiops.workbench.utils.mappers.FirecloudMapper;
 import org.pmiops.workbench.workspaces.WorkspaceAuthService;
+import org.pmiops.workbench.workspaces.resources.WorkspaceResourceMapper;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
 public class UserMetricsController implements UserMetricsApiDelegate {
 
-  private static final Logger logger = Logger.getLogger(UserMetricsController.class.getName());
-
   private static final int MAX_RECENT_NOTEBOOKS = 8;
+
   private static final Logger log = Logger.getLogger(UserMetricsController.class.getName());
+
+  private final CloudStorageClient cloudStorageClient;
+  private final FireCloudService fireCloudService;
   private final Provider<DbUser> userProvider;
   private final UserRecentResourceService userRecentResourceService;
-  private final WorkspaceDao workspaceDao;
   private final WorkspaceAuthService workspaceAuthService;
-  private final FireCloudService fireCloudService;
-  private final CloudStorageClient cloudStorageClient;
-  private final CommonMappers commonMappers;
-  private FirecloudMapper firecloudMapper;
-  private int distinctWorkspaceLimit = 5;
-
-  // TODO(jaycarlton): migrate these private functions to MapStruct
-  // Converts DB model to client Model
-  // RW-5638
-  private final Function<DbUserRecentResource, WorkspaceResource> TO_CLIENT =
-      userRecentResource -> {
-        WorkspaceResource resource = new WorkspaceResource();
-        resource.setCohort(TO_CLIENT_COHORT.apply(userRecentResource.getCohort()));
-        resource.setConceptSet(TO_CLIENT_CONCEPT_SET.apply(userRecentResource.getConceptSet()));
-        FileDetail fileDetail = convertStringToFileDetail(userRecentResource.getNotebookName());
```

Comment:
this is the only line that's not a straightforward assignment or conversion.  This is now in WorkspaceResourceMapper.